src/roiextractors/memmapextractors.py

```diff
@@ -2,109 +2,118 @@
 import shutil
 import tempfile
 from pathlib import Path
+from typing import Union, Optional
 
 import numpy as np
 from tqdm import tqdm
 
 from .extraction_tools import PathType, check_get_frames_args
 from .imagingextractor import ImagingExtractor
 
+PathType = Union[str, Path]
+DtypeType = Union[str, np.dtype]
+OptionalDtypeType = Optional[DtypeType]
+DtypeType = Union[str, np.dtype]
+
 
 class MemmapImagingExtractor(ImagingExtractor):
+
+    extractor_name = "MemmapImagingExtractor"
+
     def __init__(
         self,
-        imaging_extractor: ImagingExtractor,
-        save_path: PathType = None,
-        verbose: bool = False,
+        file_path: PathType,
+        frame_shape: tuple,
+        dtype: DtypeType,
+        offset: int = 0,
+        sampling_frequency: float = 0,
+        image_structure_to_axis: dict = None,
```

Comment:
Similarly, can be `Dict[str, int]` to better indicate expected format of specification

Reply:
Thanks!